File: dev_server.py
import http.server
import socketserver
import os

PORT = 5501

class MyHandler(http.server.SimpleHTTPRequestHandler):
    def end_headers(self):
        self.send_header('Cross-Origin-Opener-Policy', 'same-origin')
        self.send_header('Cross-Origin-Embedder-Policy', 'require-corp')
        self.send_header('Cross-Origin-Resource-Policy', 'same-origin')
        self.send_header('Cache-Control', 'no-store, no-cache, must-revalidate')
        super().end_headers()

    def do_GET(self):
        print(f"Request: {self.path}")
        return super().do_GET()

os.chdir(os.path.dirname(os.path.abspath(__file__)))
print(f"Serving files from: {os.getcwd()}")

with socketserver.TCPServer(("", PORT), MyHandler) as httpd:
    print(f"Server started at http://localhost:{PORT}")
    httpd.serve_forever()

File: dev_server.js
const express = require('express');
const app = express();
const port = 5500;

app.use((req, res, next) => {
  res.setHeader('Cross-Origin-Opener-Policy', 'same-origin');
  res.setHeader('Cross-Origin-Embedder-Policy', 'require-corp');
  res.setHeader('Cross-Origin-Resource-Policy', 'same-origin');
  next();
});

app.use(express.static('.'));

app.listen(port, () => {
  console.log(`Training Server running at http://localhost:${port}`);
});
